modules/percolator/src/test/java/org/elasticsearch/percolator/PercolatorQuerySearchIT.java

```diff
@@ -71,6 +75,14 @@
 
 public class PercolatorQuerySearchIT extends ESIntegTestCase {
 
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        List<Class<? extends Plugin>> plugins = new ArrayList<>();
+        plugins.addAll(super.nodePlugins());
+        plugins.add(PercolatorPlugin.class);
+        return plugins;
+    }
+
```

Comment:
Is this change needed for this pr ? It seems unrelated.

Reply:
The problem is that MockNioTransport does not register [NamedWriteables](https://github.com/elastic/elasticsearch/blob/95d46332f8c5e52e731fabcfee1c78056f0fa38d/test/framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java#L116) from the [bundled plugins](https://github.com/elastic/elasticsearch/blob/95d46332f8c5e52e731fabcfee1c78056f0fa38d/server/src/main/java/org/elasticsearch/node/Node.java#L400). We need to manually register the percolate plugin in the test as we now de/serialize the Percolate QueryBuilder between search phases. I looked into how to fix this issue, but it should be in a follow-up.